examples/cql-time-types.rs

```diff
@@ -40,19 +40,16 @@ async fn main() -> Result<()> {
         )
         .await?;
 
-    if let Some(rows) = session
+    let result = session
         .query("SELECT d from examples_ks.dates", &[])
-        .await?
-        .rows
-    {
-        for row in rows.into_typed::<(NaiveDate,)>() {
-            let (read_date,): (NaiveDate,) = match row {
-                Ok(read_date) => read_date,
-                Err(_) => continue, // We might read a date that does not fit in NaiveDate, skip it
-            };
-
-            println!("Parsed a date into chrono::NaiveDate: {:?}", read_date);
-        }
+        .await?;
+    for row in result.rows_typed::<(NaiveDate,)>()? {
+        let (read_date,): (NaiveDate,) = match row {
+            Ok(read_date) => read_date,
+            Err(_) => continue, // We might read a date that does not fit in NaiveDate, skip it
+        };
+
+        println!("Parsed a date into chrono::NaiveDate: {:?}", read_date);
     }
```

Comment:
The previous semantics was nonsense IMO. We highly expect here to get `Rows` response. In case we get something different than `Rows`, we should return with an error rather than continue.